projects/demo/src/modules/customization/wrapper/examples/1/index.ts

```diff
@@ -8,8 +12,14 @@ import {TUI_TEXTFIELD_APPEARANCE, tuiCheckboxOptionsProvider} from '@taiga-ui/co
     encapsulation: ViewEncapsulation.None,
     providers: [
         {
-            provide: TUI_TEXTFIELD_APPEARANCE,
-            useValue: 'material-textfield',
+            provide: TUI_TEXTFIELD_APPEARANCE_DIRECTIVE,
```

Comment:
weren't you gonna drop that token in v4?

Reply:
or did you mean?

```ts
{
  provide: TUI_TEXTFIELD_APPEARANCE_DIRECTIVE,
  useValue: {
    appearance: 'material-textfield',
  }
}
```

anyway, I prefer `useFactory` over `useValue` because you can specify a return type. with `useValue` I can make a mistake (e.g. write appear**e**nce) and it won't complain, but won't work in a runtime either. since Angular providers are poorly typed (thank god they partially fixed it with `inject()`), I'd rather make sure everything is ok myself

but in this case I've got only the directive for return type, I can't return just an object. we can fix it by introducing an interface, which the directive will implement, then we'll be able to return just that interface for simple cases like this one

but if we do this, I feel like this should be a separate task, as it makes sense to do it for all the textfield directives at once, not just for the appearance

Reply:
All textfield directives currently already behave this way, but yeah, it's kind of undocumented in types. I just don't want to spook people with complicated examples, that's why I would prefer it to be simple. Interface and types are a separate task, sure. Something is wrong with build, though. Seems like some real cyclic dependency now.

Reply:
changed the example, and hopefully fixed the build error